Feature/ae 1815 kayttaja api key hash

muep opened the pull request.

muep committed:
AE-1815 Move api_key_hash from laatija to kayttaja

This commit is only the database layout part - more refactoring will
follow

muep committed:
AE-1815 Take api_key_hash to whoami from kayttaja

muep committed:
AE-1815 Move :api-key from laatija schema to kayttaja

muep committed:
AE-1815 Change tests to expect the schema changes

muep committed:
AE-1815 Change services to have api keys in kayttaja

hkorpi reviewed (approved):
:+1:

hkorpi commented on etp-db/src/main/sql/migration/v5-release-1.1/v5.33-kayttaja-api-key-hash.sql:
```diff
@@ -0,0 +1,14 @@
+from laatija
+where kayttaja.id = laatija.id;
+
+alter table laatija drop column api_key_hash;
```
tämän voisi tehdä myöhemmin

muep committed:
AE-1815 Postpone removal of api_key_hash from laatija

This improves compatibility of the new layout with old code, which
gets momentarily run with the new layout during deployment of the
upgrade.

The removal can be done afterwards in a separate changeset.

muep commented on etp-db/src/main/sql/migration/v5-release-1.1/v5.33-kayttaja-api-key-hash.sql:
```diff
@@ -0,0 +1,10 @@
+where kayttaja.id = laatija.id;
+
+alter table audit.kayttaja add column api_key_hash text;
+
```
Tähän lisäksi:
```
drop trigger if exists kayttaja_update_trigger on kayttaja;
call audit.create_audit_procedure('kayttaja'::name);
call audit.create_audit_update_trigger('kayttaja'::name, 'kayttaja'::name,
  audit.update_condition('kayttaja'::name));
```

muep committed:
AE-1815 Refresh audit triggers on kayttaja table

This also remedies a missing trigger on the valvoja column.
